wezterm-gui/src/spawn.rs

```diff
@@ -68,17 +68,26 @@ pub async fn spawn_command_internal(
         None
     };
 
-    let cmd_builder = if let Some(args) = spawn.args {
-        let mut builder = CommandBuilder::from_argv(args.iter().map(Into::into).collect());
-        for (k, v) in spawn.set_environment_variables.iter() {
-            builder.env(k, v);
-        }
-        if let Some(cwd) = spawn.cwd {
-            builder.cwd(cwd);
+    let cmd_builder = match (
+        spawn.args.as_ref(),
+        spawn.cwd.as_ref(),
+        !spawn.set_environment_variables.is_empty(),
+    ) {
+        (None, None, false) => None,
```

Comment:
Negative logic can be harder to grok, let's flip this around so that it is a bit easier to follow!

```suggestion
        spawn.set_environment_variables.is_empty(),
    ) {
        (None, None, true) => None,
```